fsharp-backend/src/BackendOnlyStdLib/HttpBytesClient.fs

```diff
@@ -0,0 +1,212 @@
+            let query =
+              // Remove the question mark
+              if uri.Query.Length > 0 then String.dropLeft 1 uri.Query else uri.Query
+            HttpQueryEncoding.createQueryString query queryParams
+
+          System.UriBuilder(
+            Scheme = uri.Scheme,
+            Host = uri.Host,
+            Port = uri.Port,
+            Path = uri.AbsolutePath,
+            Query = queryString
+          )
+
+        use req =
+          new HttpRequestMessage(
+            method,
+            string reqUri,
+            Version = System.Net.HttpVersion.Version30,
+            Content = new ByteArrayContent(reqBody)
+          )
+
+        // headers
+        // TODO is it still appropriate to include these default headers?
+        let defaultHeaders =
+          Map [ "Accept", "*/*"; "Accept-Encoding", "deflate, gzip, br" ]
+
+        Map reqHeaders
+        |> Map.mergeFavoringRight defaultHeaders
+        |> Map.iter (fun k v ->
+          // TODO do we need this branch?
+          // I think so - .NET is odd with content types, to my memory
+          if String.equalsCaseInsensitive k "content-type" then
+            try
+              req.Content.Headers.ContentType <-
+                Headers.MediaTypeHeaderValue.Parse(v)
+            with
+            | :? System.FormatException ->
+              Exception.raiseCode "Invalid content-type header"
+          else
+            // Dark headers can only be added once, as they use a Dict.
+            // Remove them so they don't get added twice.
+            // TODO: re-evaluate if we need this. See logic in HttpClient.
+            // Also, consider that we could (do, in this draft) return headers
+            // as a list of tuples, so the Dict claim is false currently.
+            req.Headers.Remove(k) |> ignore<bool>
+            let added = req.Headers.TryAddWithoutValidation(k, v)
+
+            // Headers are split between req.Headers and req.Content.Headers so just try both
+            if not added then
+              req.Content.Headers.Remove(k) |> ignore<bool>
+              req.Content.Headers.Add(k, v))
+
+        // send request
+        Telemetry.addTag "request.content_type" req.Content.Headers.ContentType
+        Telemetry.addTag "request.content_length" req.Content.Headers.ContentLength
+        use! response = httpClient.SendAsync req
+
+        Telemetry.addTags [ "response.status_code", response.StatusCode
+                            "response.version", response.Version ]
+        use! responseStream = response.Content.ReadAsStreamAsync()
+        use memoryStream = new MemoryStream()
+        do! responseStream.CopyToAsync(memoryStream)
+        let respBody = memoryStream.ToArray()
+
+        let headers =
+          response
+          |> HttpHeaders.headersForAspNetResponse
+          // TODO Should we really be manipulating (lowercasing) these?
+          |> List.map (fun (k : string, v) -> (String.toLowercase k, v))
```

Comment:
Should we also lowercase the values?

Reply:
I don't understand why we should lowercase _either_, in request or response. Could you please explain? Why is it the library fn's job to adjust the casing?